cmd/hype/cli/cli.go

```diff
@@ -197,24 +197,23 @@ func WithTimeout(ctx context.Context, timeout time.Duration, f func(context.Cont
 	}
 
 	cltx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
+	errCh := make(chan error, 1)
 
-	cltx, cause := context.WithCancelCause(cltx)
+	defer cancel()
 
 	go func() {
-		defer cancel()
-		err := f(cltx)
-		if err != nil {
-			cause(err)
-		}
+		// TODO: check for context cancellation in f() to prevent leaking the goroutine
```

Comment:
Is this still a TODO or was this fixed?

Reply:
Theoretically, there is a scenario when the goroutine executing the callback in `WithTimeout()` is leaked. It would happen if the callback doesn't terminate before the timeout, and the context is cancelled. To prevent leaking the goroutine, the callback function should monitor the context and return when it's cancelled. I might be wrong, and I only skimmed through the code, but it seems like this is not the case. However, this is not critical since the error from canceling the context on timeout is lifted all the way up anyway, and the application exits with the respective exit code.